crates/functional/src/lib.rs

```diff
@@ -0,0 +1,456 @@
+use std::borrow::Borrow;
+use std::cell::RefCell;
+use std::ops::DerefMut;
+use std::rc::Rc;
+use yew::{Component, ComponentLink, Html, Properties};
+
+thread_local! {
+    static CURRENT_HOOK: Rc<RefCell<Option<HookState>>> = Rc::new(RefCell::new(None));
```

Comment:
no need for `Rc` wrapper right?

Reply:
I need a persistent copy of the state to keep for later use

Reply:
I don't really understand, isn't it possible to remove the `Rc` wrapper here?

Reply:
If we have them pushing to next_tick, they need to have a way to access the RefCell :thinking:

Reply:
Ah, it's because of `next_tick`? Didn't realize that! It's fine then 👍